lib/apiservers/engine/backends/image.go

```diff
@@ -179,3 +193,76 @@ func convertV1ImageToDockerImage(image *metadata.ImageConfig) *types.Image {
 		Labels:      labels,
 	}
 }
+
+// Retrieve the image metadata from the storage port layer.
+func getImageConfigFromCache(image string) (*metadata.ImageConfig, error) {
+	// Use docker reference code to validate the id
+
+	digest, named, err := reference.ParseIDOrReference(image)
+	if err != nil {
+		return nil, err
+	}
+
+	// Try to get the image config from our image cache
+	imageCache := ImageCache()
+
+	if digest != "" {
+		config := imageCache.GetImageByDigest(digest)
+		if config != nil {
+			return config, nil
+		}
+	} else {
+		config := imageCache.GetImageByNamed(named)
+		if config != nil {
+			return config, nil
+		}
+	}
+
+	return nil, derr.NewRequestNotFoundError(fmt.Errorf("No such image: %s", image))
+}
+
+// Converts the data structure retrieved from the portlayer.  This src datastructure
+// represents the unmarshalled data saved in the storage port layer.  The return
+// data is what the Docker CLI understand and returns to user.
+func imageConfigToDockerImageInspect(imageConfig *metadata.ImageConfig, productName string) *types.ImageInspect {
```

Comment:
We can add some validation.  The question is where?  My personal preference is to add the validation in the image cache on update.  That way, all operations that require image data can trust that source data.